siliconcompiler/core.py

```diff
@@ -2771,9 +2773,101 @@ def _find_leaves(self, steplist):
         # Then, find all leaf tasks by elimination.
         return all_tasks.difference(non_leaf_tasks)
 
+    ###########################################################################
+    def _generate_summary_image(self, input_path, output_path):
+        '''Takes a layout screenshot and generates a design summary image
+        featuring a layout thumbnail and several metrics.'''
+
+        # Extract metrics for display
+
+        metrics = {
+            'Chip': self.get('design'),
+        }
+
+        if self.get('option', 'pdk'):
+            metrics['Node'] = self.get('option', 'pdk')
+
+        # TODO: a bit hardcoded to asicflow assumptions... a way to query
+        # "final" metrics regardless of flow would be handy
+        totalarea = self.get('metric', 'totalarea', step='export', index='1')
+        if totalarea:
+            metrics['Area'] = f'{int(totalarea)} um^2'
+
+        fmax = self.get('metric', 'fmax', step='export', index='1')
+        if fmax:
+            # SI-ify
+            if fmax < 1e6:
+                fmax = f'{round(fmax / 1e3, 2)} KHz'
+            elif fmax < 1e9:
+                fmax = f'{round(fmax / 1e6, 2)} MHz'
+            else:
+                fmax = f'{round(fmax / 1e9, 2)} GHz'
+            metrics['Fmax'] = fmax
+
+        # Generate design
+
+        WIDTH = 512
+        BORDER = 16
+        LINE_SPACING = 4
+        TEXT_INDENT = 8
+
+        FONT_PATH = os.path.join(self.scroot, 'data', 'RobotoMono', 'RobotoMono-Regular.ttf')
```

Comment:
Do subfolders get included?

Reply:
apparently this breaks our current setup.py config (blocking CI). Should be fixable though